### PR TITLE
Location permissions set

### PR DESCRIPTION
**Pull request by juanpablo-pereira**
## Getting permission for location
- ### Asking for permission of location access done
- ### Display of the availability of the permission
location-permission-gif

### CODE REVIEW

File: android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt [hunk 1]
```diff
@@ -1,30 +1,139 @@
...
+            allowed = ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED && ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED
+            permissionEventsSource?.success(
+                if (allowed) {
+//                    resources.getText(R.string.permission_allowed)
```
Comment (FranciscoGaraventa) [inline android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt]: Remove this comment line

File: android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt [hunk 2]
```diff
@@ -1,30 +1,139 @@
...
+                    permissionAllowed
+                } else {
+//                    resources.getText(R.string.permission_not_allowed)
```
Comment (FranciscoGaraventa) [inline android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt]: Remove this comment line

File: lib/another_location_plugin.dart [hunk 3]
```diff
@@ -5,9 +5,40 @@ import 'package:flutter/services.dart';
...
+  static Stream<dynamic> permissionStream =
+      _permissionEventChannel.receiveBroadcastStream();
 
   static Future<String?> get platformVersion async {
     final String? version = await _channel.invokeMethod('getPlatformVersion');
     return version;
   }
+
+  static Future<bool> get initializePlugin async {
+    final bool isInitialized = await _channel.invokeMethod('initializePlugin');
```
Comment (FranciscoGaraventa) [inline lib/another_location_plugin.dart]: You can add this String and the others than appears here into a String file

File: android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt (hunk 1 above)
Comment (juanpablo-pereira) [reply android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt]: Removed!

File: android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt (hunk 2 above)
Comment (juanpablo-pereira) [reply android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt]: Also removed!

File: lib/another_location_plugin.dart (hunk 3 above)
Comment (juanpablo-pereira) [reply lib/another_location_plugin.dart]: Added to utils/Strings file

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM

File: android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt [hunk 4]
```diff
@@ -1,30 +1,137 @@
...
-  private lateinit var channel : MethodChannel
+class AnotherLocationPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
+    PluginRegistry.ActivityResultListener {
+    private lateinit var channel: MethodChannel
 
-  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "another_location_plugin")
-    channel.setMethodCallHandler(this)
-  }
+    private var permissionEventsEmitter: EventChannel? = null
```
Comment (MarcosCGlobant) [inline android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt]: avoid setting variables as null, you can use lateinit var here

File: android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt [hunk 5]
```diff
@@ -1,30 +1,137 @@
...
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
-    } else {
-      result.notImplemented()
+            override fun onCancel(arguments: Any?) {
+                permissionEventsSource = null
+            }
+        }
+    private var context: Context? = null
+    private var activity: Activity? = null
```
Comment (MarcosCGlobant) [inline android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt]: same as above, it shouldn't be set to null

File: android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt [hunk 6]
```diff
@@ -1,30 +1,137 @@
...
+
+    companion object Constants {
+        const val pluginChannelName = "another_location_plugin"
+        const val permissionChannelName = "permission_event_channel"
+        const val getPlatformVersion = "getPlatformVersion"
+        const val checkPermissions = "checkPermissions"
+        const val requestPermissions = "requestPermissions"
+        const val permissionAllowed = "Permission allowed"
+        const val permissionNotAllowed = "Permission not allowed"
+        const val requestCode = 500
```
Comment (MarcosCGlobant) [inline android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt]: the name mangling for the companion objects uses  MAYUS and the words separated by lower hyphen: 
 e.g. PLUGIN_CHANNEL_NAME

File: example/lib/main.dart [hunk 7]
```diff
@@ -22,21 +25,16 @@ class _MyAppState extends State<MyApp> {
...
   Future<void> initPlatformState() async {
     String platformVersion;
-    // Platform messages may fail, so we use a try/catch PlatformException.
-    // We also handle the message potentially returning null.
+
     try {
-      platformVersion =
-          await AnotherLocationPlugin.platformVersion ?? 'Unknown platform version';
+      platformVersion = await AnotherLocationPlugin.platformVersion ??
+          'Unknown platform version';
```
Comment (MarcosCGlobant) [inline example/lib/main.dart]: you can add this to the String file

File: android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt (hunk 4 above)
Comment (juanpablo-pereira) [reply android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt]: Done!

File: android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt (hunk 5 above)
Comment (juanpablo-pereira) [reply android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt]: Done

File: android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt (hunk 6 above)
Comment (juanpablo-pereira) [reply android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt]: Understand, done!

File: example/lib/main.dart (hunk 7 above)
Comment (juanpablo-pereira) [reply example/lib/main.dart]: Added

**Pull request merged**